glue/sales_val_flagging.py

```diff
@@ -523,15 +512,47 @@ def modify_dtypes(df):
     for col in df.select_dtypes("Int64").columns:
         df[col] = df[col].astype("int64")
 
+    # Function to ensure all numeric values are converted to float
+    def to_float(value):
+        if isinstance(value, list):
+            return [float(i) for i in value]
+        elif isinstance(value, np.ndarray):
+            return value.astype(float).tolist()
+        elif isinstance(value, (int, float, np.integer, np.floating)):
+            return float(value)
+        else:
+            return value
+
+    # Apply conversion function to specified columns
+    conversion_columns = ["dev_bounds", "ptax_sd"]
+    for col in conversion_columns:
+        if col in df.columns:
+            df[col] = df[col].apply(to_float)
+
+    # Adjustments for specific string columns, ensuring compatibility
+    string_columns = [
+        "run_id",
+        "run_filter",
+        "iso_forest_cols",
+        "stat_groups",
+        "tri_stat_groups",
+        "housing_market_class_codes",
+    ]
+    for col in string_columns:
+        if col in df.columns:
+            df[col] = df[col].astype(str)
+
```

Comment:
This change helps to standardize dtypes, it is built for the `sale.parameter` table. This addition solves two problems:

- String encoded json columns. This is a hard check that all of the columns in `string_columns` are changed to string type before being written to athena. This check helps to make sure athena dtypes are consistent so there are no issues when building the tables
- Nested `array<double>` problems. There was a problem with the `ptax_sd` column data. In some runs the values in the python list that gets written to athena are float, and in others the values are int. When a python list is written to athena, the resulting dtype is `array<$NUMERIC_DTYPE>`.  Since athena can't handle compound types, it tries to translate the entire `array` to either double or int. This resulted in different parquets having a different athena dtype which cause hive errors upon using athena to query the table. The code in the function manually changes all of the elements within the lists to float, so that no matter what athena will interpret them as `array<double>`
- Another problem was found in the process of reading prod data, manipulating it, and re-writing it. The fix for the prior issue works as we look inside a list and change the dtype of the elements inside the list. A problem with this approach was that when reading existing parquet data from prod, these `array` columns are read back into python as numpy arrays. Manually changing the elements of a list does not work for the numpy array data structure, so we manually change any numpy array to a list so that the int --> float conversion works in all scenarios.